pkg/R/RDD.R

```diff
@@ -1063,7 +1078,7 @@ setMethod("saveAsObjectFile",
           function(rdd, path) {
             # If the RDD is in string format, need to serialize it before saving it because when
             # objectFile() is invoked to load the saved file, only serialized format is assumed.
-            if (!rdd@env$serialized) {
+            if (rdd@env$serialized == "string") {
```

Comment:
I've tried it both ways and it doesn't look like simply calling `reserialize` on a "row"-serialized RDD will do the trick. Since all `reserialize` does is recreate the PipelinedRDD via an lapply, I don't think the "row" RDD actually changes at all when you run try to reserialize it.